sympy/core/expr.py

```diff
@@ -822,19 +821,17 @@ def _eval_is_negative(self):
             # 34 bits of accuracy
             except ValueError:
                 return None
-            if n2 is None:
+            if n2 is None or n2 is S.NaN:
                 return None
             if getattr(n2, '_prec', 1) == 1:  # no significance
                 return None
-            if n2 == S.NaN:
-                return None
 
-            n, i = self.evalf(2).as_real_imag()
-            if not i.is_Number or not n.is_Number:
+            r, i = n2.as_real_imag()
+            if not i.is_Number or not r.is_Number:
                 return False
-            if n._prec != 1 and i._prec != 1:
-                return bool(not i and n < 0)
-            elif n._prec == 1 and (not i or i._prec == 1) and \
+            if r._prec != 1 and i._prec != 1:
+                return bool(not i and op(n))
```

Comment:
Should this be `op(n2)`?

Reply:
no
```suggestion
                return bool(not i and op(r))
```

Reply:
Good eye, @jksuom -- it was formerly `n` but I changed it to `r` to avoid confusion with `n2`.